### PR TITLE
Splitting 'show' and 'clear' parse-chains to address overlapping functionality

### PR DESCRIPTION
**Pull request by rodnymolina**
The following patch provides a simple mechanism to allow overlapping parse-chains to co-exist without interfering with each other. In this particular example, we are separating BGP functionality attending to the routing-stack being used in the system. As part of these changes, Quagga-based systems will exclusively rely on 'show ip/ipv6 bgp' and 'sonic-clear ip/ipv6 bgp' stanzas, whereas FRR will make use of 'show bgp ipv4/ipv6' and 'sonic-clear bgp ipv4/ipv6' ones. Also, a few bugs are getting fixed here.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -37,6 +38,7 @@
             'config = config.main:cli',
             'sfputil = sfputil.main:cli',
```
Comment (jleveque) [inline setup.py]: I believe you also need to add entry points here under `console_scripts` for `debug` and `undebug`

File: MANIFEST.in [hunk 2]
```diff
@@ -1 +1,2 @@
+include *.py
```
Comment (jleveque) [inline MANIFEST.in]: Why is this line needed? I believe we should be able to specify all needed .py files via setup.py, either via packages or scripts.

File: setup.py (hunk 1 above)
Comment (rodnymolina) [reply setup.py]: Right, will add those two right away.

File: MANIFEST.in (hunk 2 above)
Comment (rodnymolina) [reply MANIFEST.in]: That's part of a long story. I had a very hard time trying to add the new bgp_*.py files into a show/bgp/ and clear/bgp subfolders. I gave up at the end and i left my new files at the same level as main.py. This added line, reflects part of my effort tweaking packages' building-config to add the new bgp folder to the .deb package. Even though i was able to incorporate the new folders into the generated python-sonic-util.tar.gz, these ones never made it into the python-sonic-util.deb package. I gave up at the end, and i decided to remove the bgp folders, so I will go ahead and remove this line now.

File: show/bgp_frr_v4.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+def advertised_routes(ipaddress):
+    command = 'sudo vtysh -c "show bgp ipv4 neighbor {} advertised-routes"'.format(ipaddress)
+    run_command(command)
+
+
+# 'received-routes' subcommand ("show bgp neighbors received-routes <nbr>")
+@neighbors.command('received-routes')
+@click.argument('ipaddress')
+def received_routes(ipaddress):
+    command = 'sudo vtysh -c "show bgp ipv4 neighbor {} received-routes"'.format(ipaddress)
```
Comment (lguohan) [inline show/bgp_frr_v4.py]: this is not supported, because we did not enable route cache in quagga. use show bgp ipv4 neighbor {} routes instead.

File: show/main.py [hunk 4]
```diff
@@ -455,6 +465,14 @@ def syseeprom():
     run_command("sudo decode-syseeprom")
 
 
+# 'hardware' command ("show hardware")
+@cli.command()
+def hardware():
+    """Show local hardware information"""
+    command = "sudo dmidecode -t 1,2,10"
```
Comment (lguohan) [inline show/main.py]: do we have this dmidecode in the base image? also, which dmidecode is related to hardware. This is not requirement for platform support, there is no guarantee that this will be supported by all sonic platforms.

File: show/bgp_frr_v4.py (hunk 3 above)
Comment (rodnymolina) [reply show/bgp_frr_v4.py]: Good catch. Will remove this command for the time being.

File: show/main.py (hunk 4 above)
Comment (rodnymolina) [reply show/main.py]: No, it's not in base-image, i was planning to add it though. However, i just noticed that you guys recently introduced "show platform", which overlaps with the info provided by this command, so i will remove this one from my patch.

**Comment by rodnymolina** [comment]
I have taken care of all the comments posted above. Please let me know if there's anything else missing.

File: clear/main.py [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+
+
+CONTEXT_SETTINGS = dict(help_option_names=['-h', '--help', '-?'])
+
+
+#
+# 'cli' group (root group) ###
+#
+
+# THis is our entrypoint - the main "Clear" command
```
Comment (jleveque) [inline clear/main.py]: `THis` -> `This` (lowercase 'h').

Comment (rodnymolina) [reply clear/main.py]: done

File: clear/bgp_frr_v4.py [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+    else:
+        command = 'sudo vtysh -c "clear bgp *"'
+    run_command(command)
+
+
+# 'in' subcommand
+@neighbor.command('in')
+@click.argument('ipaddress', required=False)
+def neigh_in(ipaddress):
+    if ipaddress is not None:
```
Comment (jleveque) [inline clear/bgp_frr_v4.py]: Many of these commands and groups are missing docstrings. The Click library uses the docstrings to present usage information to the user as part of the 'help' text.

Could you please go through all files and add all missing docstrings; just a brief description of each group/command?

Test each group/command by using the `--help` flag.

**Comment by lguohan** [comment]
code looks good to me.

**Comment by stcheng** [comment]
resolve the conflicts before merging?

**Comment by jleveque** [comment]
Still conflicted?

**Comment by rodnymolina** [comment]
i used github's 'resolve-conflicts' tab directly to fix the conflict (see '359c1fc' above), but not sure we have enough 'rights' to edit the code through this channel. Are you still seeing conflicts?

**Comment by jleveque** [comment]
Yes. Still seeing conflicts here. I've never used the Github interface for resolving conflicts, so I'm not sure what might have gone wrong.

**Comment by jleveque** [comment]
Retest this please.
